backend/utils/oauth_utils.py

```diff
@@ -36,6 +36,12 @@ def get_or_create_user(user_info):
     username = user_info["login"]
     email = user_info["email"]
 
+    filterUser = User.objects.filter(email=email).first()
+    if filterUser:
+        if filterUser.id42 is None:
```

Comment:
este if y el de arriba deberían ser uno

Reply:
eso y lo que tienes es lo mismo. Lo que me refiero es a:

```python
if filterUser and filterUser.id42 is None:
  return None
```
Y ya gestionas fuera lo que sea. Tal como lo tienes ahora rompe el flujo del get_or_create

Reply:
-El metodo first() si no encuentra nada te devuelve un none que es lo que gestiono.

- la segunda pregunta no la entiendo muy por oauth siempre te va a venir un email y un usuario correcto si el correo ya esta en la base de datos pero el usuario es otro te devuelvo un error que es el de la imagen.

- luego probare la logica porque quizás pueda ser un create solo ahora ya que si existe el id42 ya lo devuelvo antes.

Reply:
- Si first te devuelve None y con poner ```if filterUser``` ya te vale, entonces el ```if filterUser.id42 is None``` tendría que ser ````if filterUser.id42``` o el otro "if filterUser is None". Si none es lo mismo que null en c, entonces la primera opcion,
- La segunda es preguntarte como se gestionan los datos. Puesto que me dices que ya esta todo controlado y que supuestamente no puede haber problemas con los datos de entrada, entonces el segundo filtrado por nombre de usuario te sobra ya que ya tienes el usuario al filtrar por email
- Pues eso te he dicho en el comentario anterior, si ya estas gestionando el get, ya no tiene que ser un get_or_create, sino un create

A ver si con esto ya lo terminas, que es innecesaria tanta conversacion pa un cambio de 3 lineas

Reply:
- El is None hace lo mismo que un == NULL en c o un ! cuando se niega en un if(!user), lo que hago es
        if filterUser: -> si existe entra al if
        if filterUser.id42 is None: -> si no existe (= a None) entra al if
            return None
        return filterUser

-Si cuando te metes oauth va con la info de la cuenta del 42, por lo tanto el email como el usuario tiene que estar bien.
Luego el tema del segundo if es necesario se encarga de validar si existe ese username en la bbdd igual que antes se verifica el email ahora el username en caso de que existe se le añade _42 para luego crear el usuario por ejemplo si existe en la bbdd atrujill y me logeo yo se me creara con el nombre atrujill_42

-Correcto no es necesario el get ya porque lo devuelvo antes en caso de que venga el id42 asi que ya subo ese cambio :)

Reply:
No me expliques lo que hace el codigo. Te estoy diciendo que:
1) quites lo del is None y pongas el not correspondiente, al igual que has obviado la condicion explicita en el otro if
2) que te ahorres la segunda llamada al filter
3) y que separes ese doble if en dos bloques independientes

Reply:
Podríamos resolver esto juntos en una llamada? Yo no estoy entendiendo esta conversación tan larga por un if...

Reply:
Sino, yo lo dejaría como está y ya, si está probado y funciona como debería, no hay problema, tampoco es una locura de código, y no tenemos que hacerlo ni que sea lo más eficiente de universo, ni que sean todas las practicas perfectas y el mínimo nº de ifs ni nada.

Con que Ale haya aprendido dos o 3 cosas nuevas ya va guay.

Que corrija lo del create con el default y ya está, yo considero que el trabajo está bien hecho